service/frontend/workflowHandler.go

```diff
@@ -1337,8 +1338,9 @@ func (wh *WorkflowHandler) GetWorkflowExecutionHistory(
 		return nil, err
 	}
 
-	if getRequest.GetMaximumPageSize() == 0 {
-		getRequest.MaximumPageSize = common.Int32Ptr(wh.config.DefaultHistoryMaxPageSize)
+	if getRequest.GetMaximumPageSize() <= 0 {
+		getRequest.MaximumPageSize = common.Int32Ptr(int32(
```

Comment:
This is still too verbose.  Can we have a strongly typed API on config like:
GetHistoryMaxPageSize(domain)?

Reply:
Refactored